### ISSUE TITLE
bug: old and new versions do not use the same deps

### ISSUE THREAD

**Issue by monperrus** [revapi/revapi#102]
We have a strange behavior with Revapi. It [now detects](https://github.com/INRIA/spoon/pull/1632#issuecomment-340215520) changes related to generics as follows. However, those changes are spurious.

    old: method java.util.ArrayList Main::handleBootclasspath(ArrayList, java.lang.String) @ JDTBatchCompiler
    new: method java.util.ArrayList<String> Main::handleBootclasspath(ArrayList<String>, java.lang.String) @ JDTBatchCompiler

The difference is only in the generics.

It seems that:

* either the old version does not contain the generics information
* or Revapi does not detect does not detect the generics information in the old version
* or Revapi does not compute the type erasure

Since the old and new versions are not compiled on the same machine, this may be the reason.

How are generics compiled?  how are generics detected by Revapi? How to solve this problem?

**Comment by metlos** [comment]
`mvn package revapi:check -X` reveals:

```
Old API:
API[archives=[fr.inria.gforge.spoon:spoon-core:jar:6.0.0-20171030.103854-124], supplementary=[org.eclipse.tycho:org.eclipse.jdt.core:jar:3.12.0.v20160516-2131, commons-io:commons-io:jar:1.4, ant:ant:jar:1.6.5, org.apache.openejb:javaee-api:jar:5.0-2, log4j:log4j:jar:1.2.17, com.martiansoftware:jsap:jar:2.1]]

New API:
API[archives=[fr.inria.gforge.spoon:spoon-core:jar:6.0.0-SNAPSHOT], supplementary=[org.apache.openejb:javaee-api:jar:5.0-2, commons-io:commons-io:jar:2.5, log4j:log4j:jar:1.2.17, org.apache.maven:maven-model:jar:3.5.0, org.eclipse.tycho:org.eclipse.jdt.core:jar:3.13.50.v20171007-0855, com.martiansoftware:jsap:jar:2.1, org.apache.commons:commons-lang3:jar:3.5, org.codehaus.plexus:plexus-utils:jar:3.0.24]]
```
I'm still not sure why, but the reason here is that the old version is taken from `org.eclipse.tycho:org.eclipse.jdt.core:jar:3.12.0.v20160516-2131`. Not sure where this gets resolved from or why...

**Comment by monperrus** [comment]
That's a good diagnostic, thanks

The POM of `fr.inria.gforge.spoon:spoon-core:jar:6.0.0-20171030.103854-124` points to the correct Tycho/JDT version.

Why does Revapi get the wrong dependency version? (`eclipse.jdt.core:jar:3.12.0.v20160516-2131` instead of `.eclipse.jdt.core:jar:3.13.50.v20171007-0855`)

(Ill try to flush the Tycho Maven cache)

**Comment by monperrus** [comment]
Flushing the Tycho Maven cache does not help. Interestingly, the maven runs downloads the two versions:

```
drwxr-xr-x 2 gakoci gakoci 4096 Oct 31 09:35 3.13.50.v20171007-0855
drwxr-xr-x 2 gakoci gakoci 4096 Oct 31 09:35 3.12.0.v20160516-2131
```

**Comment by metlos** [comment]
Are these two artifacts also downloaded when revapi is not running in the build?

**Comment by monperrus** [comment]
> Are these two artifacts also downloaded when revapi is not running in the build?

It seems not. Flushing the Tycho Maven cache + `mvn clean package -DskipTests -X ` only downloads one artifact.

**Comment by metlos** [comment]
So it seems like the bug is indeed in how revapi uses maven to resolve the dependencies. Interesting no one encountered this bug yet... Thanks for reporting!

**Comment by monperrus** [comment]
And more interesting, this bug has appeared recently, only when we changed the version of Tycho/JDT-core

**Comment by metlos** [comment]
I just released `revapi-maven-plugin-0.9.5` which should fix this issue... (god, I love being able to fix a bug and send out a new release while on the train :) ).

**Comment by monperrus** [comment]
I confirm it works now! We can re-enable Revapi!

Thanks a lot for the fast work!
